src/openrct2-ui/windows/Staff.cpp

```diff
@@ -119,1240 +123,1147 @@ static rct_widget* window_staff_page_widgets[] = {
+            PeepPickupType::Place, staffEntityId, { destCoords, tileElement->GetBaseZ() }, network_get_current_player_id()
+        };
+        pickupAction.SetCallback([](const GameAction* ga, const GameActions::Result* result) {
+            if (result->Error != GameActions::Status::Ok)
+                return;
+            tool_cancel();
+            gPickupPeepImage = ImageId();
+        });
+        GameActions::Execute(&pickupAction);
+    }
 
-    window_align_tabs(w, WIDX_TAB_1, WIDX_TAB_3);
-}
+    void OverviewToolAbort(rct_widgetindex widgetIndex)
+    {
+        if (widgetIndex != WIDX_PICKUP)
+            return;
 
-/**
- *
- *  rct2: 0x006BDD91
- */
-void WindowStaffOverviewInvalidate(rct_window* w)
-{
-    ColourSchemeUpdateByClass(w, static_cast<rct_windowclass>(WC_STAFF));
+        PeepPickupAction pickupAction{ PeepPickupType::Cancel,
+                                       EntityId::FromUnderlying(number),
+                                       { picked_peep_old_x, 0, 0 },
+                                       network_get_current_player_id() };
+        GameActions::Execute(&pickupAction);
+    }
 
-    if (window_staff_page_widgets[w->page] != w->widgets)
+    void OverviewViewportRotate()
     {
-        w->widgets = window_staff_page_widgets[w->page];
-        WindowInitScrollWidgets(w);
+        ViewportInit();
     }
 
-    w->pressed_widgets |= 1ULL << (w->page + WIDX_TAB_1);
-
-    const auto peep = GetStaff(w);
-    if (peep == nullptr)
+    void OverviewTextInput(rct_widgetindex widgetIndex, std::string_view text)
     {
-        return;
-    }
-    auto ft = Formatter::Common();
-    peep->FormatNameTo(ft);
+        if (widgetIndex != WIDX_RENAME)
+            return;
 
-    window_staff_overview_widgets[WIDX_BACKGROUND].right = w->width - 1;
-    window_staff_overview_widgets[WIDX_BACKGROUND].bottom = w->height - 1;
+        if (text.empty())
+            return;
 
-    window_staff_overview_widgets[WIDX_RESIZE].right = w->width - 1;
-    window_staff_overview_widgets[WIDX_RESIZE].bottom = w->height - 1;
+        auto gameAction = StaffSetNameAction(EntityId::FromUnderlying(number), std::string{ text });
+        GameActions::Execute(&gameAction);
+    }
+#pragma endregion
 
-    window_staff_overview_widgets[WIDX_TITLE].right = w->width - 2;
+#pragma region Options tab events
+    void OptionsMouseUp(rct_widgetindex widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case WIDX_CHECKBOX_1:
+            case WIDX_CHECKBOX_2:
+            case WIDX_CHECKBOX_3:
+            case WIDX_CHECKBOX_4:
+                SetOrder(widgetIndex - WIDX_CHECKBOX_1);
+                break;
+        }
+    }
 
-    window_staff_overview_widgets[WIDX_VIEWPORT].right = w->width - 26;
-    window_staff_overview_widgets[WIDX_VIEWPORT].bottom = w->height - 14;
+    void OptionsOnMouseDown(rct_widgetindex widgetIndex)
+    {
+        rct_widget* widget = &widgets[widgetIndex];
```

Comment:
As we only care about the dropdown widget just get that instead.
```
auto* ddWidget = &widgets[WIDX_COSTUME_BOX];
```
then you can drop the -- and the comment.